packages/manager/apps/pci-private-registry/src/pages/create/steps/RegionStep.tsx

```diff
@@ -52,12 +68,45 @@ export default function RegionStep({
     return [];
   }, [localisations, capabilities]);
 
+  const getDeploymentZones = (acc: TDeployment[], region: TLocalisation) => {
+    const deploymentZone = acc.find((zone) => zone.name === region.type);
+    if (!deploymentZone)
+      acc.push({
+        name: region.type,
+      });
```

Comment:
You could use Object.groupBy or a Set instead which would take less lines of code

Reply:
the problem with the set is that it adds all the objects because they are compared by reference...

for groupBy, I'd have to switch to ES2024 (which could be a good thing) but I don't think it would be more readable...
If you have a code suggestion that I haven't thought of, don't hesitate, otherwise we could leave it as it is.

What do you think?